src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -55,6 +55,12 @@ def register_options(cls, register):
       advanced=True,
       help='The maximum timeout (in seconds) that can be set on a test target.',
     )
+    register(
+      '--coverage',
```

Comment:
In future, rather than defining this on the test-backend-specific Subsystem, it would be good to see whether it is possible to define it generically as part of the `test` goal, and to then inject coverage options (likely via #7490). It would be good to see a followup PR (or a design doc) so that we can talk about how to prepare for doing this generically, if possible.

Reply:
All the coverage options will end up on a coverage specific subsystem I think. 

I'm not sure how much coverage options will overlap across languages as I've not used them outside python.

Reply:
I think we should aim to have a unified coverage interface where possible, so I agree with having `--coverage` defined on the `test` `Goal`, and plumbing it through, erroring if you try to run it for a language we don't support.  It would be wonderful to generate unified reports for different languages in a consistent format, if possible.

Reply:
Do we currently support coverage for any other language? What are those interfaces like? Agreed they should be unified if possible.

Reply:
Yes, we currently support coverage for Java, Scala, and Python.

It sounds like you incorporated this feedback into the design. Thank you!

Reply:
> Yes, we currently support coverage for Java, Scala, and Python.

in V2?

Reply:
No, in v1.